CMake/utility.cmake

```diff
@@ -386,6 +386,54 @@ function(activate_precompiled_headers _PRECOMPILED_HEADER _SOURCE_FILES)
   endif()
 endfunction()
 
+####################################################################################
+# Adds warnings compiler options to the target depending on the category
+# _TARGET Target name
+# _CATEGORY "library" or "unit_tests" or "benchmarks"
+####################################################################################
+macro( add_warning_levels _TARGET _CATEGORY)
+  if("${_CATEGORY}" STREQUAL "library" OR "${_CATEGORY}" STREQUAL "examples")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(CXX_FLAGS "-Wall" "-Werror")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      set(CXX_FLAGS "-Wall" "-Werror")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+      set(CXX_FLAGS "/WX" "/W4")
+    else()
+      message(FATAL_ERROR "Unknown compiler to set warning levels: ${_CATEGORY} for target ${_TARGET}! ")
+    endif()
+  elseif("${_CATEGORY}" STREQUAL "unit_tests")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(CXX_FLAGS "-Wall" "-Werror" "-Wno-unused-private-field" "-Wno-delete-non-virtual-dtor")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      set(CXX_FLAGS "-Wall" "-Werror" "-Wno-unused-private-field" "-Wno-delete-non-virtual-dtor")
+      if( ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER "5.0")
+        list(APPEND CXX_FLAGS "-Wno-noexcept-type")
```

Comment:
I think this can be removed right, because you fixed something in #107 : 
What happens when you build this library with a compiler which was not aware of the noexcept for the calling convention, and then link against it with modern code compiled with c++17...?

Reply:
I think, you should always link & compile with the same compiler flags and options.
There is no ABI in C++, so it might not load the library anyway.